packages/@lwc/engine-core/src/framework/queue-microtask.ts

```diff
@@ -0,0 +1,44 @@
+import { ArrayPush, isFunction, globalThis } from '@lwc/shared';
+
+type Callback = () => void;
+
+/* istanbul ignore next */
+function queueMicrotaskPolyfill() {
+    let nextTickCallbackQueue: Callback[] = [];
+
+    function flushCallbackQueue() {
+        if (process.env.NODE_ENV !== 'production') {
+            if (nextTickCallbackQueue.length === 0) {
+                throw new Error(
+                    `Internal Error: If callbackQueue is scheduled, it is because there must be at least one callback on this pending queue.`
+                );
+            }
+        }
+        const callbacks: Callback[] = nextTickCallbackQueue;
+        nextTickCallbackQueue = []; // reset to a new queue
+        for (let i = 0, len = callbacks.length; i < len; i += 1) {
+            callbacks[i]();
+        }
+    }
+
+    return function addCallbackToNextTick(callback: Callback) {
+        if (process.env.NODE_ENV !== 'production') {
+            if (!isFunction(callback)) {
+                throw new Error(
+                    `Internal Error: addCallbackToNextTick() can only accept a function callback`
+                );
+            }
+        }
+        if (nextTickCallbackQueue.length === 0) {
+            Promise.resolve().then(flushCallbackQueue);
+        }
+        ArrayPush.call(nextTickCallbackQueue, callback);
+    };
+}
+
+// This is supported in all modern browsers https://developer.mozilla.org/en-US/docs/Web/API/queueMicrotask#browser_compatibility
+// Basically the fallback is just for IE11
```

Comment:
By explicitly mentioning IE11 here, I'm trying to make this easier to find when we drop IE11 support.